lib/good_migrations/logic.rb

```diff
@@ -1,5 +1,35 @@
 module GoodMigrations
-  class PreventsAppLoad
+  class Logic
+    def self.currently_executing_migration_path
+      migrate_dir_path = Rails.root.join("db/migrate/").to_s
+      loc = caller.detect { |loc| loc.start_with?(migrate_dir_path) }
+      return if loc.nil?
+      loc.partition(":").first
+    end
+
+    def self.extract_time_from_migration_path(path)
+      timestamp_string = File.basename(path).partition("_").first
+      return if timestamp_string.size != 14
+      Time.parse(timestamp_string)
+    end
+
+    def self.permit_autoloading_based_on_migration_time?
+      return false if GoodMigrations::Configuration.permit_autoloading_before_date.nil?
+
+      migration_path = currently_executing_migration_path
+      return false if migration_path.nil?
+
+      migration_time = extract_time_from_migration_path(migration_path)
+      # Or should this raise?
+      return false if migration_time.nil?
+
+      migration_time < GoodMigrations::Configuration.permit_autoloading_before_date
+    end
```

Comment:
I'm not comfortable with this level of complexity and mutation being added as a series of class methods

Reply:
I extracted this to a MigrationDetails class